ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/input/KSTInputFBControllerStateEstimator.java

```diff
@@ -168,8 +184,273 @@ public FrameVector3DReadOnly getEstimatedCoMVelocity()
       return centerOfMassEstimator.getEstimatedVelocity();
    }
 
-   private class SingleEndEffectorEstimator
+   private interface SingleEndEffectorStateEstimatorBase
+   {
+
+      void reset();
+
+      void update(double time,
+                  double defaultLinearRateLimitation,
+                  double defaultAngularRateLimitation,
+                  long inputTimestamp,
+                  KinematicsToolboxRigidBodyCommand input);
+
+      void predict(double time);
+
+      FramePose3DReadOnly getEstimatedPose();
+
+      SpatialVectorReadOnly getEstimatedVelocity();
+   }
+
+   private class SingleEndEffectorEstimatorValentine implements SingleEndEffectorStateEstimatorBase
```

Comment:
Let's rename this estimator to `SingleEndEffectorC1Estimator` and we'll rename the other one to `SingleEndEffectorC0Estimator`